library/bignum_core.c

```diff
@@ -746,7 +746,93 @@ static void exp_mod_precompute_window(const mbedtls_mpi_uint *A,
     }
 }
 
+#if defined(MBEDTLS_TEST_HOOKS) && !defined(MBEDTLS_THREADING_C)
+// Set to a default that is neither MBEDTLS_MPI_IS_PUBLIC nor MBEDTLS_MPI_IS_SECRET
+int mbedtls_mpi_optionally_safe_codepath = MBEDTLS_MPI_IS_PUBLIC + MBEDTLS_MPI_IS_SECRET + 1;
```

Comment:
I don't like this. A + B + 1 is not a good way to get a number that's neither A nor B - what if A = 0 and B = -1? I know that's not true here, but this expression has a dependency on the values of A and B (and we don't have a static assert that this value = neither of them).

How would you feel about a third value specified in the #defines, of something like MBEDTLS_IS_NETHER_FOR_TESTING (name is horrible, but you get the idea)?

Reply:
I'm unresolving this as we've reverted past the commit that addressed this

Reply:
I've already added this commit to the [followup PR](https://github.com/Mbed-TLS/mbedtls/pull/9493)